fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/DefaultLoanLifecycleStateMachine.java

```diff
@@ -62,7 +65,7 @@ private LoanStatus getNextState(LoanEvent loanEvent, Loan loan) {
         }
 
         LoanStatus from = loan.status();
-        LoanStatus newState = from;
+        LoanStatus newState = null;
```

Comment:
Why was this needed, could you please explain?

Reply:
To avoid sending out from "ACTIVE" to "ACTIVE" events. Either we are not indicating with the result that there was no state change or we need to check the before & after and decide based on that.